yaptide/simulation_runner/shieldhit_runner.py

```diff
@@ -10,108 +10,39 @@
-*---><---><--------><------------------------------------------------>
-    0    0           protons, H2O 30 cm cylinder, r=10, 1 zone
-*---><---><--------><--------><--------><--------><--------><-------->
-  RCC    1       0.0       0.0       0.0       0.0       0.0      30.0
-                10.0
-  RCC    2       0.0       0.0      -5.0       0.0       0.0      35.0
-                15.0
-  RCC    3       0.0       0.0     -10.0       0.0       0.0      40.0
-                20.0
-  END
-  001          +1
-  002          +2     -1
-  003          +3     -2
-  END
-* material codes: 1 - liquid water (ICRU material no 276), 1000 - vacuum, 0 - black body
-    1    2    3
-    1 1000    0
-"""
-
-
-def run_shieldhit(param_dict):
+def run_shieldhit(param_dict, raw_input_dict):
     """Shieldhit runner"""
-    import pymchelper
-
-    input_dict = input_cfg_templ.copy()
-
-    input_dict['beam.dat'] = input_dict['beam.dat'].format(
-        energy=param_dict['energy'],
-        nstat=param_dict['nstat']
-    )
-
-    input_dict['detect.dat'] = input_dict['detect.dat'].format(
-        cyl_nr=param_dict['cyl_nr'],
-        cyl_nz=param_dict['cyl_nz'],
-        mesh_nx=param_dict['mesh_nx'],
-        mesh_ny=param_dict['mesh_ny'],
-        mesh_nz=param_dict['mesh_nz']
-    )
-
     # create temporary directory
     with tempfile.TemporaryDirectory() as tmp_output_path:
 
-        for config_filename in input_dict:
-            abs_input_path = os.path.join(tmp_output_path, config_filename)
-            with open(abs_input_path, 'w') as temp_input_file:
-                temp_input_file.write(input_dict[config_filename])
+        conv_runner = ConvRunner(parser=DummmyParser(),
+                                 input_data=raw_input_dict,
+                                 output_dir=tmp_output_path)
+
+        conv_runner.run_parser()
 
+        print(os.listdir(tmp_output_path))
         settings = SimulationSettings(input_path=tmp_output_path,
                                       simulator_exec_path=None,
                                       cmdline_opts='')
 
-        runner_obj = Runner(jobs=param_dict['jobs'],
-                            keep_workspace_after_run=False,
-                            output_directory=tmp_output_path)
+        runner_obj = SHRunner(jobs=param_dict['jobs'],
+                              keep_workspace_after_run=False,
+                              output_directory=tmp_output_path)
 
         start_time = timeit.default_timer()
-        isRunOk = runner_obj.run(settings=settings)
-        if not isRunOk:
+        if not runner_obj.run(settings=settings):
             return None
 
         elapsed = timeit.default_timer() - start_time
         print("MC simulation took {:.3f} seconds".format(elapsed))
 
         estimator = runner_obj.get_data()
 
-        return estimator
-
-
-if __name__ == '__main__':
-    sys.exit(run_shieldhit(sys.argv[1:]))
+        return estimator['mesh_']
```

Comment:
We have now configuration generated from JSON file, corenames of the output files may largely differ, depending on what user provides in the JSON file. Why do you assume a fixed value of `mesh_` ?

Reply:
changed in #104